lib/spack/spack/fetch_strategy.py

```diff
@@ -684,16 +699,22 @@ def fetch(self):
         if self.commit:
             # Need to do a regular clone and check out everything if
             # they asked for a particular commit.
-            args = ['clone', self.url, self.stage.source_path]
-            if not spack.config.get('config:debug'):
-                args.insert(1, '--quiet')
-            git(*args)
+            debug = spack.config.get('config:debug')
```

Comment:
Why not use the git destination option here (as Todd wanted)?

Reply:
Sorry I'm not familiar with that (the option or the context) - is this from a discussion in #11528?

Reply:
Yes, see https://github.com/spack/spack/pull/11528#discussion_r287420897.

Reply:
I think the question is "why not check out the repository out directly to the `spack-src` directory?". The issue with that is that when we do that, we lose the repository name:  for example if we do `git clone https://github.com/spack` then we get a directory called `spack`; this is useful when choosing a directory name for the archive in two cases:

* This avoids putting resources named `spack-src` into a root stage
* This avoids caching an SCM package into a directory called `spack-src` (i.e. when placed in `var/spack/cache/`)

Does that seem reasonable?

Reply:
Ah.  Thanks for the clarification.